### PR TITLE
add code demo

### PR DESCRIPTION
**Pull request by jinghanjia**

### CODE REVIEW

File: code/Attack.ipynb [hunk 1]
```diff
@@ -0,0 +1,1620 @@
+{
```
Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Before starting the cell, provide a brief description of what the readers will be doing in this notebook. Considering including any important details and references. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Give a run down of what we're downloading here. Also, the kind of models being evaluated here. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Let's move this installation block at the beginning. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='6'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>We don't know anything about the dataset. Provide a bride description. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Consider including:</p><p> </p><ul><li>Code comments.</li><li>What the preprocessing steps are. </li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Same as above:</p><p> </p><ul><li>What are we preprocessing.</li><li>Code comments. </li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='9'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Consider including a brief workflow of the evaluation pipeline we're following. It's unclear. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>What should readers expect after they run this code cell? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='12'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>The images are not rendering. Could you please check? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='25'/>

**Comment by sayakpaul** [comment]
@jinghanjia thanks for the PR. 

A common comment is on the lack of descriptions in the notebook. In its current form, it's hard to follow the code and understand what it's doing. Please include descriptions where needed.

**Comment by sayakpaul** [comment]
@jinghanjia thanks for the updates, I will look into those ASAP. Could you also provide me with a recording where you are walking through the code in notebook (at least the major parts)?

File: code/README.md [hunk 2]
```diff
@@ -1 +1,24 @@
...
\ No newline at end of file
+This directory will contain code (notebooks / scripts) for evaluate robustness of programming language models that are based on Transformers [1].
+
+## Code navigation
+
+This directory provide the notebook for evaluating robustness of programming language models by using two  attack generation methods [2,3] in code domain. We focus on code summarization task, and provide three models to compare their performance. 
+
+* Supervised training .
+* Contrastive learning (Contracode) [4].
+* Robustness-aware Contrastive learning (CLAW)[tbd].
```
Comment (sayakpaul) [inline code/README.md]: If we're using it, could you please cite it appropriately?

File: code/README.md [hunk 3]
```diff
@@ -1 +1,24 @@
...
+
+ 
+
+## References
```
Comment (sayakpaul) [inline code/README.md:13]: Could you comment on the images? What do they represent? Maybe include short descriptions for each of the images?

Comment (sayakpaul) [reply code/README.md:13]: Also, the optimal attach is optimal because it just changes a single function argument w.r.t the clean example. Is it so?

**Review by sayakpaul — CHANGES_REQUESTED** [review]

File: code/Attack.ipynb [hunk 4]
```diff
@@ -0,0 +1,1466 @@
...
```
Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Provide a one-liner description of code summarization. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Include descriptions for each of the artifacts. We don't know which Transforer model(s) is being used. Which test set is being used, and so on. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='4'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>What are we seeing after " + Loaded 19999 samples "?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>What is a program sketch? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='9'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Which model is being used here? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='11'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Can you provide a visual example of an example generated using this method would look like? And also provide the reference that introduced this attack. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='14'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>References. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='21'/>

File: code/Attack.ipynb [hunk 5]
```diff
@@ -0,0 +1,1471 @@
...
```
Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Is there a paper that discusses the training method, the architecture, etc? Additionally, which dataset was used to train the model? Please include these details. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Reference missing. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Which method is used to generate the adversarial examples? Include details. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>CLAW reference missing. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (sayakpaul) [inline code/Attack.ipynb:1]: <p>Which method was used for generating the adversarial examples? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/robustness-foundation-models/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (jinghanjia) [reply code/Attack.ipynb:1]: <p>I haven't put this paper on the arxiv, will put it on later. Then I will add the reference .</p>

Comment (sayakpaul) [reply code/Attack.ipynb:1]: Then at least leave a placeholder so that we don't forget.
